src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+
+
+_processed_filename = "biomasse_{iso3}_{admin}_dekad_{start_dekad}.csv"
+
+
+def _get_processed_path(admin, start_dekad, iso3: Union[None, str] = None):
+    if iso3 is None:
+        iso3 = "general"
+
+    _processed_path = os.path.join(
+        os.getenv("AA_DATA_DIR"), "public", "processed", iso3, "biomasse"
+    )
+
+    return os.path.join(
+        _processed_path,
+        _processed_filename.format(
+            iso3=iso3, admin=admin, start_dekad=start_dekad
+        ),
+    )
+
+
+AdminArgument = Literal["ADM0", "ADM1", "ADM2"]
+_valid_admin = get_args(AdminArgument)
+
+
+def _check_admin(admin):
+    if admin not in _valid_admin:
+        raise ValueError("`admin` must be one of 'ADM0', 'ADM1', or 'ADM2'.")
+
+
+def download_dmp(admin: AdminArgument = "ADM2"):
+    """Download raw DMP data
+
+    Raw DMP data is downloaded for specified
+    administrative area. The data is downloaded
+    in CSV format and can be processed using
+    ``process_dmp()``.
+    """
+    _check_admin(admin)
+    url = _base_url.format(admin=admin)
+    raw_path = _get_raw_path(admin)
+    download_ftp(url=url, save_path=raw_path)
+    return None
+
+
+def load_dmp(admin: AdminArgument = "ADM2", redownload: bool = False):
+    _check_admin(admin)
+    raw_path = _get_raw_path(admin)
+    if not os.path.exists(raw_path):
+        download_dmp(admin)
+    elif redownload:
+        os.remove(raw_path)
+        download_dmp(admin)
+
+    df = pd.read_csv(raw_path, na_values=["-9998.8"])
+    df.dropna(axis="columns", how="all", inplace=True)
+    return df
+
+
+def process_dmp(
+    admin: AdminArgument = "ADM2",
+    redownload: bool = False,
+    start_dekad: int = 10,
+):
+    """Process DMP raw data
+
+    DMP raw data is received in a wide format
+    dataset that needs processing. This pivots
+    the data from wide to long format with
+    DMP dekadal observations for all years, while
+    also pivoting the mean values to allow for
+    calculation of biomasse anomaly.
+    """
+    df = load_dmp(admin, redownload)
+
+    # process mean and DMP separately since mean values
+    # are dekadal and DMP are year/dekadal
+    # keep ID column for working with multipolygon admin areas
+    df_mean = df.filter(regex="(^admin|^DMP_MEA|^ID)")
+    df_dmp = df.filter(regex="(^admin|^DMP_[0-9]+|^ID)")
+    admin_cols = [col for col in df_mean.columns if col.startswith("admin")]
+
+    # groupby to average out for the few cases where admin areas
+    # are not contiguous
+    df_mean_long = (
+        pd.wide_to_long(
+            df_mean,
+            i=admin_cols + ["IDBIOHYDRO"],
+            j="dekad",
+            stubnames="DMP_MEA",
+            sep="_",
+        )
+        .reset_index()
+        .drop(labels="IDBIOHYDRO", axis=1)
+        .groupby(admin_cols + ["dekad"])
+        .mean()
+        .reset_index()
+    )
+
+    # calculate anomaly for mean separate from
+    # observed to ensure unique dekadal values
```

Comment:
this comment is vague to me. Season index is basically a shifted dekad number such that 1 corresponds to the start of the season right?